api/tests/routes/pro/patch_offer_test.py

```diff
@@ -59,6 +63,58 @@ def test_withdrawal_can_be_updated(self, client):
 
+    @override_features(WIP_ENABLE_WITHDRAWAL_UPDATED_MAIL=True)
+    def test_withdrawal_update_send_email_to_each_related_booker(self, client):
+        # given
+        offer = offers_factories.OfferFactory(subcategoryId=subcategories.CONCERT.id)
+        offerers_factories.UserOffererFactory(user__email="user@example.com", offerer=offer.venue.managingOfferer)
+        stock = offers_factories.StockFactory(offer=offer)
+        bookings = [bookings_factories.BookingFactory(stock=stock) for _ in range(3)]
+
+        data = {
+            "withdrawalDetails": "conditions de retrait",
+            "withdrawalType": "no_ticket",
+            "shouldSendMail": "true",
+        }
+        with mock.patch("pcapi.core.mails.transactional.send_booking_withdrawal_updated") as mailer_mock:
+            # when
+            response = client.with_session_auth("user@example.com").patch(f"/offers/{humanize(offer.id)}", json=data)
+
+        # then
+        assert response.status_code == 200
+        assert mailer_mock.call_count == 3
+        args_list = sorted([call.kwargs for call in mailer_mock.call_args_list], key=itemgetter("offer_token"))
+        bookings.sort(key=lambda b: b.activationCode.code if getattr(b, "activationCode") else b.token)
+        assert [args["recipients"] for args in args_list] == [[b.user.email] for b in bookings]
+        assert [args["user_first_name"] for args in args_list] == [b.user.firstName for b in bookings]
+        assert [args["offer_name"] for args in args_list] == [b.stock.offer.name for b in bookings]
+        assert [args["offer_token"] for args in args_list] == [
+            b.activationCode.code if b.activationCode else b.token for b in bookings
+        ]
+        assert [args["offer_withdrawal_delay"] for args in args_list] == [None] * 3
+        assert [args["offer_withdrawal_details"] for args in args_list] == ["conditions de retrait"] * 3
+        assert [args["offer_withdrawal_type"] for args in args_list] == ["no_ticket"] * 3
+        assert [args["offerer_name"] for args in args_list] == [offer.venue.managingOfferer.name] * 3
+        assert [args["venue_address"] for args in args_list] == [offer.venue.address] * 3
+
+    @override_features(WIP_ENABLE_WITHDRAWAL_UPDATED_MAIL=True)
+    def test_withdrawal_update_does_not_send_email_if_not_specified_so(self, client):
+        offer = offers_factories.OfferFactory(subcategoryId=subcategories.CONCERT.id)
+        offerers_factories.UserOffererFactory(user__email="user@example.com", offerer=offer.venue.managingOfferer)
+        stock = offers_factories.StockFactory(offer=offer)
+        _ = [bookings_factories.BookingFactory(stock=stock) for _ in range(3)]
+
+        data = {
+            "withdrawalDetails": "conditions de retrait",
+            "withdrawalType": "no_ticket",
+            "shouldSendMail": "false",
+        }
+        with mock.patch("pcapi.core.mails.transactional.send_booking_withdrawal_updated") as mailer_mock:
```

Comment:
<!-- Issues represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 issue:** ‏Pas besoin de mock ici
L'email ne sera pas envoyé et sera à la place récupéré dans une outbox

On peut faire 
```python
import pcapi.core.mails.testing as mails_testing

assert len(mails_testing.outbox) == 1
email_data = mails_testing.outbox[0]

# assert stuff on email_data
```